### ISSUE TITLE
[Spike] Consider Adding PHPStan to Static Tests of ECE-Tools Code

### ISSUE THREAD

**Issue by YPyltiai** [magento/ece-tools#508]
Internal ID: `MAGECLOUD-3578`

**Background:**
* https://github.com/phpstan/phpstan
* PHPStan focuses on finding errors in your code without actually running it. It catches whole classes of bugs even before you write tests for the code. It moves PHP closer to compiled languages in the sense that the correctness of each line of the code can be checked before you run the actual line.

**To Do:**
* Consider Adding PHPStan to Static Tests of ECE-Tools Code
* Exclude Test directory from the code analyze
```
parameters:
	excludes_analyse:
		- %rootDir%/../../../tests/*/data/*
```
* Run analyse command and fix all failed tests
* Add PHPStan into Travis

**Comment by YPyltiai** [comment]
@bbatsche , @shiftedreality , did we decide anything on this one? Should we proceed / close?

**Comment by shiftedreality** [comment]
That makes sense for ECE-Tools

**Comment by YPyltiai** [comment]
Resolved in #531
